server/schedule/operator_controller.go

```diff
@@ -362,26 +427,30 @@ func (oc *OperatorController) addOperatorLocked(op *operator.Operator) {
 
 	heap.Push(&oc.opNotifierQueue, &operatorWithTime{op: op, time: oc.getNextPushOperatorTime(step, time.Now())})
 	operatorCounter.WithLabelValues(op.Desc(), "create").Inc()
+	return true
 }
 
 // RemoveOperator removes a operator from the running operators.
-func (oc *OperatorController) RemoveOperator(op *operator.Operator) (found bool) {
+func (oc *OperatorController) RemoveOperator(op *operator.Operator) bool {
 	oc.Lock()
-	defer oc.Unlock()
-	return oc.removeOperatorLocked(op)
+	removed := oc.removeOperatorLocked(op)
+	oc.Unlock()
+	if removed {
+		if op.Cancel() {
```

Comment:
I see many times  for this: 
```
if op.Cancel() {
  log xxxx 
}
oc.buryOperator(op)
```
can we uniform to this?
```
oc.buryOperator(op, status, logEntry)
fucn buryOperator(op *Operator, status *OperatorStatus, ...attchLogEntry){
 switch status{

}
switch st{
}
}
```

Reply:
The first operator of log is a `string`,  so we will use many `buryOperator(op, "")`, which is ugly.